modules/stochastic_tools/src/samplers/SobolSampler.C

```diff
@@ -0,0 +1,62 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "SobolSampler.h"
+
+template <>
+InputParameters
+validParams<SobolSampler>()
+{
+  InputParameters params = validParams<Sampler>();
+  params.addClassDescription("Sobol variance-based sensitivity analysis Sampler.");
+  params.addRequiredParam<unsigned int>(
+      "n_samples", "Number of Monte Carlo samples to perform for each distribution.");
+  params.addParam<std::vector<unsigned int>>("seeds",
+                                             std::vector<unsigned int>({19800624, 19801009}),
```

Comment:
OK, I see your design. You really do need two independent generators. My fear with this design is a user runs a series of distributions using either the same seed for both generators, or incrementing by one each time so that the first run is (A, B), (B, C) and so on. Both of these would result in invalid analyses. If you only take one seed and generate the other, that potential pitfall could be completely removed.